alchemy/test/cloudflare/redirect-rule.test.ts

```diff
@@ -0,0 +1,232 @@
+import { describe, expect } from "vitest";
+import { alchemy } from "../../src/alchemy.ts";
+import { createCloudflareApi } from "../../src/cloudflare/api.ts";
+import { RedirectRule } from "../../src/cloudflare/redirect-rule.ts";
+import { Zone } from "../../src/cloudflare/zone.ts";
+import { destroy } from "../../src/destroy.ts";
+import { BRANCH_PREFIX } from "../util.ts";
+
+import "../../src/test/vitest.ts";
+
+const api = await createCloudflareApi();
+
+const test = alchemy.test(import.meta, {
+  prefix: BRANCH_PREFIX,
+});
+
+describe("RedirectRule", () => {
+  // Use BRANCH_PREFIX for deterministic, non-colliding test resources
+  const testDomain = `${BRANCH_PREFIX}-redirect-test.dev`;
+
+  test("create, update, and delete wildcard redirect rule", async (scope) => {
+    let zone: Zone | undefined;
+    let redirectRule: RedirectRule | undefined;
+
+    try {
+      // Create a test zone first
+      zone = await Zone(`${testDomain}-zone`, {
+        name: testDomain,
+        type: "full",
+        jumpStart: false,
+      });
+
+      // Create a wildcard redirect rule
+      redirectRule = await RedirectRule(`${BRANCH_PREFIX}-wildcard-redirect`, {
+        zone: zone.id,
+        requestUrl: `https://*.${testDomain}/files/*`,
+        targetUrl: `https://${testDomain}/\${1}/files/\${2}`,
+        statusCode: 301,
+        preserveQueryString: true,
+      });
```

Comment:
Can we avoid using an actual zone here? Would prefer to just workers.dev if we can, but if it's not supported then ok. But we'll have to get an actual domain to test it...

Reply:
Unclear to me: https://developers.cloudflare.com/rules/transform/url-rewrite/create-api/. It _seems_ like the zone is required, but can you get a zone_id for a worker URL?